src/parser/shared/modules/resources/resourcetracker/ResourceTracker.ts

```diff
@@ -405,11 +416,43 @@ export default class ResourceTracker extends Analyzer {
       prevUpdate &&
       timestamp <= prevUpdate.timestamp + MULTI_UPDATE_BUFFER_MS;
     const beforeAmount =
-      reportedBeforeAmount !== undefined && !withinMultiUpdateBuffer
+      reportedBeforeAmount !== undefined && !withinMultiUpdateBuffer && !this.useGranularity
```

Comment:
whats the role of the `!this.useGranularity` here?

Reply:
`reportedBeforeAmount` is a rounded number, whilst the `calculatedBeforeAmount` have been changed to return non-rounded numbers with `this.useGranularity=true`, so we need to use the `calculatedBeforeAmount` if we want to have decimals/granularity.

Reply:
Added a comment 👍